pkg/cmd/pr/merge/merge.go

```diff
@@ -123,49 +123,58 @@ func mergeRun(opts *MergeOptions) error {
 		return err
-	} else if pr.State == "MERGED" {
-		err := fmt.Errorf("%s Pull request #%d (%s) was already merged", cs.Red("!"), pr.Number, pr.Title)
-		return err
 	}
 
-	mergeMethod := opts.MergeMethod
 	deleteBranch := opts.DeleteBranch
 	crossRepoPR := pr.HeadRepositoryOwner.Login != baseRepo.RepoOwner()
+	isTerminal := opts.IO.IsStdoutTTY()
+	isPRAlreadyMerged := pr.State == "MERGED"
 
-	if opts.InteractiveMode {
-		mergeMethod, deleteBranch, err = prInteractiveMerge(opts.DeleteLocalBranch, crossRepoPR)
-		if err != nil {
-			if errors.Is(err, cancelError) {
-				fmt.Fprintln(opts.IO.ErrOut, "Cancelled.")
-				return cmdutil.SilentError
+	if !isPRAlreadyMerged {
+		mergeMethod := opts.MergeMethod
+
+		if opts.InteractiveMode {
+			mergeMethod, deleteBranch, err = prInteractiveMerge(opts.DeleteLocalBranch, crossRepoPR)
+			if err != nil {
+				if errors.Is(err, cancelError) {
+					fmt.Fprintln(opts.IO.ErrOut, "Cancelled.")
+					return cmdutil.SilentError
+				}
+				return err
 			}
-			return err
 		}
-	}
 
-	var action string
-	if mergeMethod == api.PullRequestMergeMethodRebase {
-		action = "Rebased and merged"
-		err = api.PullRequestMerge(apiClient, baseRepo, pr, api.PullRequestMergeMethodRebase)
-	} else if mergeMethod == api.PullRequestMergeMethodSquash {
-		action = "Squashed and merged"
-		err = api.PullRequestMerge(apiClient, baseRepo, pr, api.PullRequestMergeMethodSquash)
-	} else if mergeMethod == api.PullRequestMergeMethodMerge {
-		action = "Merged"
-		err = api.PullRequestMerge(apiClient, baseRepo, pr, api.PullRequestMergeMethodMerge)
-	} else {
-		err = fmt.Errorf("unknown merge method (%d) used", mergeMethod)
-		return err
-	}
+		var action string
+		if mergeMethod == api.PullRequestMergeMethodRebase {
+			action = "Rebased and merged"
+			err = api.PullRequestMerge(apiClient, baseRepo, pr, api.PullRequestMergeMethodRebase)
+		} else if mergeMethod == api.PullRequestMergeMethodSquash {
+			action = "Squashed and merged"
+			err = api.PullRequestMerge(apiClient, baseRepo, pr, api.PullRequestMergeMethodSquash)
+		} else if mergeMethod == api.PullRequestMergeMethodMerge {
+			action = "Merged"
+			err = api.PullRequestMerge(apiClient, baseRepo, pr, api.PullRequestMergeMethodMerge)
+		} else {
+			err = fmt.Errorf("unknown merge method (%d) used", mergeMethod)
+			return err
+		}
 
-	if err != nil {
-		return fmt.Errorf("API call failed: %w", err)
-	}
+		if err != nil {
+			return fmt.Errorf("API call failed: %w", err)
+		}
 
-	isTerminal := opts.IO.IsStdoutTTY()
+		if isTerminal {
+			fmt.Fprintf(opts.IO.ErrOut, "%s %s pull request #%d (%s)\n", cs.Magenta("✔"), action, pr.Number, pr.Title)
+		}
+	} else {
+		err := prompt.SurveyAskOne(&survey.Confirm{
+			Message: fmt.Sprintf("Pull request #%d (%s) was already merged. Would you like to delete this local branch and switch to the default branch?", pr.Number, pr.Title),
```

Comment:
If we are not in interactive mode, this prompt should never happen. Additionally, if the user supplied `--delete-branch`, this prompt should never happen either — we can just continue in that case.

If you'd like, you could try additionally addressing #2715 in this PR! It will touch on the same logic.

Reply:
Thanks for the feedback! I'll make those adjustments and look at that other issue. I might not get to it until Monday. 

One other question I had for you is do you this the prompt message is too wordy/confusing? After I opened this PR I thought about it more and maybe something like `... Delete the branch locally and switch to default branch?` would be more appropriate.  I find my use of the word 'this' confusing when you consider merging a PR by number from another branch because in that case you aren't deleting 'this' branch.

Reply:
@dpromanko I like your proposed change with "...delete the branch locally and..." instead of the current verbiage.

Reply:
@mislav I updated this to account for interactive mode and the `-d` flag and also resolved #2715 in a separate commit df31fae.